jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -304,6 +304,8 @@ object XGBoost extends Serializable {
           " for now")
     }
     require(nWorkers > 0, "you must specify more than 0 workers")
+    require(nWorkers < trainingData.sparkContext.defaultParallelism,
+      "work number greater than defaultParallelism will be hang forever")
```

Comment:
I think it might be more useful to mention parameter by name and include the parent object for defaultParallelism, e.g. "nWorkers must be < sparkContext.defaultParallelism, otherwise the application would hang forever".